NitroxLauncher/Models/Utils/WindowsHelper.cs

```diff
@@ -55,6 +64,119 @@ internal static void RestartAsAdmin()
                 Log.Info("Can't restart the launcher as administrator, we already have permissions");
             }
         }
-    }
 
+        internal static void CheckFirewallRules()
+        {
+            try
+            {
+                CheckFirewallRules(FirewallDirection.Inbound);
+                CheckFirewallRules(FirewallDirection.Outbound);
+            }
+            catch (FileNotFoundException ex)
+            {
+                Log.Warn($"Tried to add firewall rule for program that does not exist: {ex.FileName}");
+            }
+            catch (UnauthorizedAccessException)
+            {
+                if (!Settings.Default.FirewallWarningShown)
+                {
+                    Settings.Default.FirewallWarningShown = true;
+                    Settings.Default.Save();
+                    MessageBox.Show("Try restarting the launcher as administrator or manually adding firewall rules for Nitrox programs. This warning won't be shown again.", "Error adding Windows Firewall rules", MessageBoxButton.OK, MessageBoxImage.Warning);
+                }
```

Comment:
Maybe we can remove this state setting by using LauncherNotifier.Warning(), will also allow us to redirect users to the upcoming troubleshoot page. And in this case without loosing any errors by hidding them

Reply:
Showing a toast each time isn't ideal either since you need admin for this to not error out. Once we have the proper validation page in the launcher we can notify users there.